core/templates/dev/head/pages/learner_dashboard/LearnerDashboard.js

```diff
@@ -371,25 +424,53 @@ oppia.controller('LearnerDashboard', [
             $scope.subsectionName = subsectionName;
             $scope.activityTitle = activity.title;
             $scope.remove = function() {
-              if (subsectionName ===
-                  LEARNER_DASHBOARD_SUBSECTION_I18N_IDS.EXPLORATIONS) {
-                var removeExpUrl = UrlInterpolationService.interpolateUrl(
-                  '/learnerincompleteactivityhandler/<activityType>' +
-                  '/<activityId>', {
-                    activityType: constants.ACTIVITY_TYPE_EXPLORATION,
-                    activityId: activity.id
-                  });
-                $http['delete'](removeExpUrl);
-              } else if (subsectionName ===
-                         LEARNER_DASHBOARD_SUBSECTION_I18N_IDS.COLLECTIONS) {
-                var removeCollectionUrl = (
-                  UrlInterpolationService.interpolateUrl(
-                  '/learnerincompleteactivityhandler/<activityType>' +
-                  '/<activityId>', {
-                    activityType: constants.ACTIVITY_TYPE_COLLECTION,
-                    activityId: activity.id
-                  }));
-                $http['delete'](removeCollectionUrl);
+              if (sectionNameI18nId ===
```

Comment:
I have a feeling you can find a cleaner way to do this that involves much less duplication and cyclomatic complexity. For example, activityType just depends on the subsectionName., so just do a quick calculation at the outset to see what it should be. I don't think you need a lot of branching here and it should be possible to do this with just a single call to UrlInterpolationService and a single call to $http['delete'] in the whole thing.